packages/create/src/cli/change/change-data-source.ts

```diff
@@ -0,0 +1,100 @@
+import type { Command } from 'commander';
+import type { AdpPreviewConfig, DescriptorVariant, PromptDefaults } from '@sap-ux/adp-tooling';
+import { generateChange, ChangeType, getPromptsForChangeDataSource, getManifest } from '@sap-ux/adp-tooling';
+import { getLogger, traceChanges } from '../../tracing';
+import { readFileSync, existsSync } from 'fs';
+import { join } from 'path';
+import { UI5Config } from '@sap-ux/ui5-config';
+import { promptYUIQuestions } from '../../common';
+
+let loginAttempts = 3;
+
+/**
+ * Add a new sub-command to change the data source of an adaptation project to the given command.
+ *
+ * @param {Command} cmd - The command to add the change data-source sub-command to.
+ */
+export function addChangeDataSourceCommand(cmd: Command): void {
+    cmd.command('data-source [path]')
+        .option('-s, --simulate', 'simulate only do not write or install')
+        .action(async (path, options) => {
+            await changeDataSource(path, { ...options }, !!options.simulate);
+        });
+}
+
+/**
+ * Changes the data source of an adaptation project.
+ *
+ * @param {string} basePath - The path to the adaptation project.
+ * @param {PromptDefaults} defaults - The default values for the prompts.
+ * @param {boolean} simulate - If set to true, then no files will be written to the filesystem.
+ */
+async function changeDataSource(basePath: string, defaults: PromptDefaults, simulate: boolean): Promise<void> {
+    const logger = getLogger();
+    try {
+        if (!basePath) {
+            basePath = process.cwd();
+        }
+        checkEnvironment(basePath);
+
+        const variant = getVariant(basePath);
+        const ui5Conf = await UI5Config.newInstance(readFileSync(join(basePath, 'ui5.yaml'), 'utf-8'));
```

Comment:
Please add the UI5 yaml file name as optional CLI parameter - if not provided use `ui5.yaml` because the name of the file could be anything.

Reply:
I'd recommend `-c --config` to be consistent with the ui5 tooling

Reply:
It could not be anything. We are generating this configuration file during project creation and later we expect it to be `ui5.yaml`.

Reply:
While it is the ui5.yaml during generation, every developer can change it at any time, the name of the config file is configurable for every command relying on the ui5 CLI. I, for example, usually have multiple ui5 files when testing against different systems.

Reply:
Fixed with: 110e617d071c60c49534231288f4d092dd1391b3